src/sys/expressionparser.cxx

```diff
@@ -98,6 +98,43 @@ class FieldParam : public FieldGenerator {
 private:
   std::string name; // The name of the parameter to look up
 };
+
+/// Define a new context to evaluate an expression in
+/// This is essentially a dynamic scope mechanism
+class FieldContext : public FieldGenerator {
+public:
+  using variable_list = std::list<std::pair<string, FieldGeneratorPtr>>;
```

Comment:
You probably want `std::vector` over `std::list`

Reply:
Done! Thanks